src/v/cluster/topics_frontend.cc

```diff
@@ -298,6 +301,58 @@ topics_frontend::create_non_replicable_topics(
       });
 }
 
+ss::future<std::vector<topic_result>>
+topics_frontend::dispatch_create_non_replicable_to_leader(
+  model::node_id leader,
+  std::vector<non_replicable_topic> topics,
+  model::timeout_clock::duration timeout) {
+    vlog(clusterlog.trace, "Dispatching create topics to {}", leader);
+    return _connections.local()
+      .with_node_client<cluster::controller_client_protocol>(
+        _self,
+        ss::this_shard_id(),
+        leader,
+        timeout,
+        [topics, timeout](controller_client_protocol cp) mutable {
+            return cp.create_non_replicable_topics(
```

Comment:
I don't think this change works as expected because this RPC call invokes a newly generated but non-implemented service endpoint in `cluster/controller_service.h`. Seems like you correctly edited the RPC IDL file, creating a new service endpoint however you didn't implement the method `cluster::service::create_non_replicable_topics` in our implementation of `controller_service` in `cluster/service.h`. 

I think you are thinking that this RPC call invokes `topics_frontend::create_non_replicable_topics` but that isn't the case. Its actually hitting our controller service. Here is how `create_topics` works:
```
ss::future<create_topics_reply>
service::create_topics(create_topics_request&& r, rpc::streaming_context&) {
    return ss::with_scheduling_group(
             get_scheduling_group(),
             [this, r = std::move(r)]() mutable {
                 return _topics_frontend.local().create_topics(
                   std::move(r.topics),
                   model::timeout_clock::now() + r.timeout);
```
Which is invoked by the RPC call here in `topics_frontend`
```
            return cp.create_topics(
              create_topics_request{std::move(topics), timeout},
              rpc::client_opts(model::timeout_clock::now() + timeout));
```
So you'll need to do something alone these lines for `non_replicable` topics as well.